### PR TITLE
Minor optimizations

### PR DESCRIPTION
**Pull request by cromefire**
A lot of minor optimizations

Fixed:
- unneeded variables / properties / functions
- Renamed listeners to start with "on" consistently
- ~~fixed missing `if (this.control)`~~
- Removed a few unneeded `@bind`s
- TODO where there might be possible cleanup
- An issue with the `ITextFieldProps` interface
- A `tag` prop to `Typography`
- Fixed some styling

### CODE REVIEW

**Comment by prateekbh** [comment]
sorry the description makes it a lot harder to review, please be specific about what is changed and what needs to be reviewed

**Comment by cromefire** [comment]
Hope that list helps

File: ts/IconButton/index.tsx [hunk 1]
```diff
@@ -27,8 +27,10 @@ export class IconButton extends MaterialComponent<
 
   public componentDidMount() {
     super.componentDidMount();
-    this.MDComponent = new MDCIconButtonToggle(this.control);
-    this.MDComponent.listen('MDCIconButtonToggle:change', this.onChange);
+    if (this.control) {
```
Comment (prateekbh) [inline ts/IconButton/index.tsx]: Why will this be missing?

File: ts/Icon/index.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ export class Icon<PropsType = {}, StateType = {}> extends MaterialComponent<
 
   protected materialDom(props) {
     const classes = ['material-icons'];
-    // CardActionIcon sends className
+    // CardActionIcon sends className  TODO: Isn't that handled by MaterialComponent
```
Comment (prateekbh) [inline ts/Icon/index.tsx]: That is user given class.
This is talking about https://github.com/prateekbh/preact-material-components/blob/master/ts/Card/index.tsx#L88

File: ts/Select/index.tsx [hunk 3]
```diff
@@ -64,7 +64,7 @@ export class Select extends MaterialComponent<ISelectProps, ISelectState> {
   }
 
   @bind
-  protected changed(e) {
+  protected onChange(e) {
```
Comment (prateekbh) [inline ts/Select/index.tsx]: Please keep this onChanged.
OnChange clashes with a native event

File: ts/IconButton/index.tsx (hunk 1 above)
Comment (cromefire) [reply ts/IconButton/index.tsx]: What will be missing, there's no "real" deletion it's just git being wired

Comment (prateekbh) [reply ts/IconButton/index.tsx]: Why this.control in an if?

File: ts/Select/index.tsx (hunk 3 above)
Comment (prateekbh) [reply ts/Select/index.tsx]: i guess it'll be safe to do so.

File: ts/IconButton/index.tsx (hunk 1 above)
Comment (prateekbh) [reply ts/IconButton/index.tsx]: @cromefire any issues?

File: ts/Snackbar/index.tsx [hunk 4]
```diff
@@ -27,7 +28,7 @@ export class Snackbar extends MaterialComponent<
   ISnackbarProps,
   ISnackbarState
 > {
-  public readonly isPureReactComponent = true;
+  public readonly isPureReactComponent = true; // TODO: Is that needed
```
Comment (prateekbh) [inline ts/Snackbar/index.tsx]: no please delete this

File: ts/IconButton/index.tsx (hunk 1 above)
Comment (cromefire) [reply ts/IconButton/index.tsx]: Not yet but there may be issues

Comment (cromefire) [reply ts/IconButton/index.tsx]: Cannot hurt

**Comment by cromefire** [comment]
Travis error:
```
ERROR in ./Drawer/index.js
Module not found: Error: Can't resolve '@material/drawer/persistent' in '/Users/travis/build/prateekbh/preact-material-components/Drawer'
 @ ./Drawer/index.js 24:18-56
 @ ./index.js
ERROR in ./Drawer/index.js
Module not found: Error: Can't resolve '@material/drawer/temporary' in '/Users/travis/build/prateekbh/preact-material-components/Drawer'
 @ ./Drawer/index.js 26:17-54
 @ ./index.js
```

File: ts/IconButton/index.tsx (hunk 1 above)
Comment (prateekbh) [reply ts/IconButton/index.tsx]: extra code.

**Comment by prateekbh** [comment]
yeah cuz i updated the deps, will merge this once we update the changes in the .39 version of mdc

**Comment by cromefire** [comment]
There you go

**Comment by cromefire** [comment]
Will merge it when the tests are fixed

**Comment by cromefire** [comment]
@prateekbh ready for merging

**Pull request merged**